### PR TITLE
[tr] Add Turkish localization for infrastructure-as-code.md

### PR DESCRIPTION
**Pull request by rwxdash**
### Describe your changes

- Added Turkish localization for infrastructure-as-code.md

Google Docs for reference

- [`infrastructure-as-code.md`](https://docs.google.com/document/d/13LmCPLw-FTBkBaFaZlAeYYpHABJd4gRxbxrArWjINhE)

### Checklist before opening this PR (put `x` in the checkboxes)
- [x] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [x] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

### CODE REVIEW

File: content/tr/infrastructure-as-code.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+status: Completed
+category: concept
+tags: ["infrastructure", "methodology", ""]
+---
+
+## Nedir
+
+Kod olarak altyapı _(Infrastructure as Code - IaC)_, altyapı tanımının bir veya daha fazla dosya olarak saklanması uygulamasıdır.
+Bu, servis olarak altyapının _(Infrastructure as a Service - IaaS)_ genellikle
+hell betiği veya diğer yapılandırma araçları aracılığıyla manuel olarak sağlandığı geleneksel modelin yerini alır.
```
Comment (eminalemdar) [inline content/tr/infrastructure-as-code.md:12]:
```suggestion
bir shell script veya diğer yapılandırma araçları aracılığıyla manuel olarak oluşturulduğu geleneksel modelin yerini alır.
```

File: content/tr/infrastructure-as-code.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Hangi Sorunları Çözer
+
+Uygulamaları bulut tabanlı bir şekilde geliştirmek, altyapının tek kullanımlık ve tekrarlanabilir olmasını gerektirir.
+Ayrıca bu uygulamaların insan müdahalesine gerek kalmadan, ihtiyaca bağlı olarak otomatik
+ve tekrarlanabilir bir şekilde ölçeklendirilmesi gerekir.
+Manuel provizyon, bulut tabanlı uygulamaların yanıt verme ve ölçeklendirme gereksinimlerini karşılayamaz.
```
Comment (eminalemdar) [inline content/tr/infrastructure-as-code.md:19]:
```suggestion
Altyapı kaynaklarının manuel olarak oluşturulması, bulut tabanlı uygulamaların yanıt verme ve ölçeklendirme gereksinimlerini karşılayamaz.
```

File: content/tr/infrastructure-as-code.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+Manuel altyapı değişiklikleri tekrarlanamaz, hızla ölçek sınırlarına ulaşır ve yanlış yapılandırma hatalarına neden olur.
+
+## Nasıl Yardımcı Olur
+
+Sunucular, yük dengeleyiciler ve alt ağlar gibi veri merkezi kaynaklarının kod olarak temsil edilmesi,
+altyapı ekiplerinin tüm yapılandırmalar için tek bir doğruluk kaynağına sahip olmalarını sağlar
+ve aynı zamanda sürüm kontrolü ve dağıtım stratejilerini uygulayarak veri merkezlerini
```
Comment (eminalemdar) [inline content/tr/infrastructure-as-code.md:26]:
```suggestion
ve aynı zamanda versiyon kontrolü ve dağıtım stratejilerini uygulayarak veri merkezlerini
```

**Review by eminalemdar — CHANGES_REQUESTED** [review]
Eline sağlık @rwxdash fakat direkt olarak kelime kelime çevirilerden kaçınmak daha iyi olur. Çünkü dil yapılarının farklılıklarından dolayı bazen kendi dilimize göre değişik ifadeler eklememiz daha mantıklı olacaktır.

**Review by aliok — APPROVED** [review]
LGTM
Leaving final approval to @eminalemdar

**Review by eminalemdar — APPROVED** [review]
LGTM

**Pull request merged**